idea/idea-fir/src/org/jetbrains/kotlin/idea/fir/intentions/HLAddAccessorIntentions.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.fir.intentions
+
+import com.intellij.codeInsight.intention.LowPriorityAction
+import org.jetbrains.kotlin.idea.api.applicator.HLApplicator
+import org.jetbrains.kotlin.idea.api.applicator.HLApplicatorInput
+import org.jetbrains.kotlin.idea.api.applicator.applicator
+import org.jetbrains.kotlin.idea.fir.api.AbstractHLIntention
+import org.jetbrains.kotlin.idea.fir.api.applicator.HLApplicabilityRange
+import org.jetbrains.kotlin.idea.fir.api.applicator.HLApplicatorInputProvider
+import org.jetbrains.kotlin.idea.fir.api.applicator.inputProvider
+import org.jetbrains.kotlin.idea.fir.applicators.ApplicabilityRanges
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtPropertySymbol
+import org.jetbrains.kotlin.idea.intentions.AbstractAddAccessorsIntention
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.load.java.JvmAbi
+import org.jetbrains.kotlin.name.ClassId
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.psiUtil.containingClass
+import org.jetbrains.kotlin.psi.psiUtil.containingClassOrObject
+import org.jetbrains.kotlin.psi.psiUtil.hasExpectModifier
+
+abstract class HLAddAccessorIntention(private val addGetter: Boolean, private val addSetter: Boolean) :
+    AbstractHLIntention<KtProperty, HLApplicatorInput.Empty>(KtProperty::class) {
+    // TODO: Copy and test range from AddAccessorIntentions.applicabilityRange
+    override val applicabilityRange: HLApplicabilityRange<KtProperty> = ApplicabilityRanges.SELF
+
+    override val applicator: HLApplicator<KtProperty, HLApplicatorInput.Empty> = applicator {
+        applyTo { ktProperty, _, _, editor ->
+            AbstractAddAccessorsIntention.applyTo(ktProperty, editor, addGetter, addSetter)
+        }
+        familyAndActionName(AbstractAddAccessorsIntention.createFamilyName(addGetter, addSetter))
+    }
+
+    override val inputProvider: HLApplicatorInputProvider<KtProperty, HLApplicatorInput.Empty> = inputProvider { ktProperty ->
+        if (ktProperty.isLocal || ktProperty.hasDelegate() ||
+            ktProperty.containingClass()?.isInterface() == true ||
+            ktProperty.containingClassOrObject?.hasExpectModifier() == true ||
+            ktProperty.hasModifier(KtTokens.ABSTRACT_KEYWORD) ||
+            ktProperty.hasModifier(KtTokens.LATEINIT_KEYWORD) ||
+            ktProperty.hasModifier(KtTokens.CONST_KEYWORD)
+        ) {
+            return@inputProvider null
+        }
```

Comment:
Please, move this logic to `isApplicableByPsi`. In general, if some checks can be done by psi only, it's better to move them to `isApplicableByPsi because it would allow to optimize applicability checking in future